### PR TITLE
[GITHUB-3] Adds workflow details to readme

### LINKED ISSUE #3: Add workflow documentation
We need some documentation on how to contribute and the release workflow.

### PR DESCRIPTION
**Pull request by lobsterdore**
This will close #3. I have added some detail on how
to open PRs and contribute, how to control the versions
and some general notes on the structure of the roles.

I think this will be useful so we can just point to new
contributors to the readme rather than talking them
through it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,106 @@
...
+You run `./bin/report.sh` to check the status of all sansible role repos.
+
+
+
+
+### Versioning
+
+The sansible release script will automatically increment the patch version by one 
+when it is run, it will check what the last version and simply increment it. In addition 
+to the semvar versions we always have a major and minor version, eg:
```
Comment (quater) [inline README.md]: This should be `semver` http://semver.org/

File: README.md [hunk 2]
```diff
@@ -1,13 +1,106 @@
...
+
+```
+[GITHUB-1] Fixes something
+
+A more detailed description of how this PR will close the issue at hand.
+```
+
+A good guide on commit messages is [Tim Pope's guide](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+
+Once your PR is opened Travis will test it, if all the test passed your PR will be
```
Comment (quater) [inline README.md]: change `test` to `tests`

File: README.md [hunk 3]
```diff
@@ -1,13 +1,106 @@
...
+
+
+
+## Contributing
+
+If you wish to contribute to one of our roles please open an issue in the roles repo first, 
+then create a branch named with the issue number (eg. GITHUB-1), making sure that you 
+create your branch from develop.
+
+When you are read to open a PR please squash all of your commits into one and
```
Comment (quater) [inline README.md]: change `read` to `ready`

**Review by quater — COMMENTED** [review]
Looks good, just minor changes required.

**Comment by lobsterdore** [comment]
@quater fixed those typos.

File: README.md [hunk 4]
```diff
@@ -1,13 +1,107 @@
...
+others so their Makefiles, tests and travis.yml files are somewhat out of date.
 
-Run `./bin/report.sh` to check the status of all sansible roles.
+In general though the roles follow a standardised pattern.
+
+* Tests are conducted using Vagrant locally and travis for PRs and branches
+* Proper YAML syntax should be used everywhere
+* Each role has an editorconfig file to ensure proper formating
+* The Makefile contains useful targets for running the tests
+* Three tags are present in each role: build, configure and assert
```
Comment (wojtek-oledzki) [inline README.md]: I would say
 "Two tags are present in each role: `build` and `configure`. The test.yml is using additional tag `assert`."

File: README.md [hunk 5]
```diff
@@ -1,13 +1,107 @@
...
+* The build tag handles installation and things that do not vary per environment
+* The configure tag handles config files and other things that vary per environment
+* The assert tag covers tests and is generally used for local tests and travis tests
```
Comment (wojtek-oledzki) [inline README.md]: sugestion: "The additional `assert` tag covers ..."

File: README.md [hunk 6]
```diff
@@ -1,13 +1,107 @@
...
+* Ansible-lint is used as part of the tests (pip install ansible-lint)
+* Variables all have a parent hash, this has the advantage of making it easier to read
+the vars, but has the disadvantage of requiring hash behaviour to be set to 'merge'
+
+
+
```
Comment (wojtek-oledzki) [inline README.md:29]: 4 empty lines pls

File: README.md [hunk 7]
```diff
@@ -1,13 +1,107 @@
...
+## Merges and releases
+
+Once a PR is merged into the develop branch Travis will run tests against it, once finished
+the develop branch needs to be merged into the master, at the moment this process is 
+manual and can be done like so:
+
+* Checkout this repo
+* Cd into your local checkout of the role that your PR was for
+* Run the sansible release script: `bash [PATH TO THIS REOP]/bin/release.sh
```
Comment (wojtek-oledzki) [inline README.md]: missing closing "`" at the end of the line

**Review by wojtek-oledzki — CHANGES_REQUESTED** [review]
few small changes required

**Comment by lobsterdore** [comment]
Thanks, changes have been applied.

**Review by wojtek-oledzki — APPROVED** [review]

**Pull request merged**